File: two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
         HashMap<Integer, Integer> map = new HashMap<>();
         for( int i = 0; i < nums.length; i++){
             int current = nums[i];
             int complement = target - current; 
             
             if( map.containsKey(complement) ){
                 return new int[]{map.get(complement), i};
             }
             else{
                 map.put(current,i);
             }
             
         }
        
        return new int[]{-1,-1};
    }
}

File: two-sum/two-sum.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val map = HashMap<Int, Int>()
        
        
        nums.forEachIndexed{ index, num ->
            val complement = target - num
            if(map.containsKey(complement)){
                map[complement]?.let{
                    return intArrayOf(it, index)
                }
            }
            else{
                map.put(num, index)
            }
        }
        
        return intArrayOf(-1,-1)
    }
}